sklearn/externals/_scipy/sparse/csgraph/_laplacian.py

```diff
@@ -0,0 +1,551 @@
+"""
+This file is a copy of the scipy.sparse.csgraph._laplacian module from SciPy 1.12
+
+Laplacian of a compressed-sparse graph
```

Comment:
Can you indicate the following in this header
 - the license of this code
 - the reason for vendoring the code
 - potential modifications
 
 ?

Reply:
I have added the license and reason for vendoring the code. There is no modification, and I have provided the PR link in scipy that solved the issue.